apisix/init.lua

```diff
@@ -868,6 +869,67 @@ local function healthcheck_passive(api_ctx)
 end
 
 
+function _M.status()
+    core.response.exit(200, core.json.encode({
+        status = "ok" }),
+        { ["Content-Type"] = "application/json"
+    })
+end
+
+function _M.status_ready()
+    local local_conf = core.config.local_conf()
+    local role = core.table.try_read_attr(local_conf, "deployment", "role")
+    local provider = core.table.try_read_attr(local_conf, "deployment", "role_" ..
+                                              role, "config_provider")
+    if provider == "yaml" or provider == "etcd" then
+        local status_shdict = ngx.shared["status-report"]
+        local ids = status_shdict:get_keys()
+        local error
+        local worker_count = ngx.worker.count()
+       if #ids ~= worker_count then
+            core.log.warn("worker count: ", worker_count, " but status report count: ", #ids)
+            error = "worker count: " .. ngx.worker.count() ..
+            " but status report count: " .. #ids
```

Comment:
It's a good idea to also print a list of ids to show which worker is not working properly. Just `core.json.encode()` it.

Reply:
Okay